docs/DeveloperGuide.md

```diff
@@ -6,32 +6,215 @@
+
+**Step2.** Once the first step is done, `listProjects` will call the `printProject` method from `ProjectList` with no parameter.
+
+**Step3.** `printList` will first check if the size of the private ArrayList named `projectList` is 0, which means that the arraylist is empty. If yes, it will print "You have not added any projects yet!" and return.
+
+**Step4.** It will then iterate over `projectList`, which type is `Project`. It will call `getTitle` and `getDeadline` from each array element in`the `Project` class.
+
+**Step5.** the general information of a project, i.e., its title and deadline, will be displayed.
+
+#### Todo feature
+![image info](./UmlDiagrams/Todo.png)  
+
+**Step1.** When `CommandHandler` receives a user input starting with string “todo”, it will call `toDo()` function to parse the user input into the index of the target `Project` object and one description string.  
+
+**Step2.** If the number of provided arguments is less than the required number, the `toDo()` function will throw an `IllegalCommand` exception and return.  
+
+**Step3.** If no exception is thrown, two strings will be generated by parsing the user input and the `toDo()` function will pass the two arguments to `projectList` for creating and adding a `Todo` object to a `Project` object.  
+
+**Step4.** When `projectList` receives the call, it will get the target `Project` object and pass the description string to it.  
+
+**Step5.** The `Project` object will create a `Todo` object and add it to the `ArrayList<Todo>` in `Project` class.  
+
+#### View a Project
+
+The view function makes use of ```ProjectList``` and ```CommandHandler``` classes. It is facilitated by methods within the ```Todo``` and ```Project``` classes, stored within ```ProjectList```.
+Given below is an example usage scenario and how View Project behaves at each step
+
+**Step 1.** The user executes ```view cs2113``` to view the project titled cs2113. The command is retrieved and parsed by ```handleUserInput()``` in ```CommandHandler```, which calls ```view()``` in ```CommandHandler```
+
+**Step 2.** The ```view()``` method ensures that ```ProjectList``` has been correctly initialised and the command is valid, before calling ```view()``` in ```ProjectList```
+
+**Step 3.** ```view()``` in ```ProjectList``` calls ```findProjectIndex()``` in ```ProjectList```
+
+**Step 4.** ```findProjectIndex()``` loops through all ```Project``` objects in projectList until a ```Project``` with the same title as the one entered by the user is found. The ```Project```’s index is returned to ```CommandHandler```'s implementation of ```view()```
+
+**Step 5.** ```view()``` calls the ```printDetails()``` method of the ```Project``` at the given index
+
+**Step 6.** ```printDetails()``` displays project title and deadline to the user. It iterates through the ```Todo```s in the ```todos``` attribute and prints each `Todo` by calling `Todo`’s `toString()` function
+
+![image info](./UmlDiagrams/ViewProject.jpg)
```

Comment:
1. Consider using opt rather than alt for this as there is only 1 condition (Condition can also be placed at the top of the box)
![image](https://user-images.githubusercontent.com/69310031/160964429-bc907229-b443-4482-9fb8-a8ef0e93bdf7.png)